In [1]:
import numpy as np


In [ ]:
# Import file and creat an array
taxi=np.genfromtxt("C:/Users/Pratik/Desktop/ML-AI/nyc_taxis.csv", delimiter=',', skip_header=True)

In [5]:
taxi

array([[2.016e+03, 1.000e+00, 1.000e+00, ..., 1.165e+01, 6.999e+01,
        1.000e+00],
       [2.016e+03, 1.000e+00, 1.000e+00, ..., 8.000e+00, 5.430e+01,
        1.000e+00],
       [2.016e+03, 1.000e+00, 1.000e+00, ..., 0.000e+00, 3.780e+01,
        2.000e+00],
       ...,
       [2.016e+03, 6.000e+00, 3.000e+01, ..., 5.000e+00, 6.334e+01,
        1.000e+00],
       [2.016e+03, 6.000e+00, 3.000e+01, ..., 8.950e+00, 4.475e+01,
        1.000e+00],
       [2.016e+03, 6.000e+00, 3.000e+01, ..., 0.000e+00, 5.484e+01,
        2.000e+00]])

In [6]:
#calculate the mean speed of rides
speed=taxi[:,7]/ (taxi[:,8]/3600)

In [7]:
mean_speed=speed.mean()
print(mean_speed)

32.24258580925573


In [8]:
#number of rides taken in february
rides_feb=taxi[taxi[:,1] ==2,1]

In [9]:
rides_feb

array([2., 2., 2., ..., 2., 2., 2.])

In [10]:
print(rides_feb.shape[0])

13333


In [11]:
#Number of rides when tip is more than 50
print(taxi[taxi[:,-3]> 50,-3].shape[0])

16


In [12]:
#number of rides when rides  drop was JFK Airport
print(taxi[taxi[:,6]==2,6].shape[0])

11832


# Analysis
Mean Speed = 32.24258580925573 m/h
number of rides taken in february= 13333
Number of rides when tip is more than 50= 16
number of rides when rides  drop was JFK Airport =11832
